src/ipmanager/api/models.py

```diff
@@ -1 +1,55 @@
 # Create your models here.
+from django.db.models import TextField, IntegerField, CharField, \
+  BooleanField, CASCADE, ForeignKey, UniqueConstraint
+from django_extensions.db.models import TimeStampedModel
+from enum import Enum
+from ipaddress import IPv4Network
+from django.core.validators import RegexValidator
+import re
+from django.core.exceptions import ValidationError
+
+class Group(TimeStampedModel):
+  key = CharField(
+    max_length=32, 
+    unique=True, 
+    validators=[RegexValidator(regex=r"\A[a-z][a-z\d_-]*\Z", flags=re.IGNORECASE)]
+  )
+  name = CharField(max_length=128, unique=True)
+  description = TextField(blank=True)
+  notes = TextField(blank=True)
+  export = BooleanField(default=False)
+  
+def validate_ipv4_or_cidr_address(value):
+  try :
+    IPv4Network(value)
+    return True
+  except ValueError:
+    return False
+
+class IPRange(TimeStampedModel):
+  group_id = ForeignKey(Group, on_delete=CASCADE)
```

Comment:
Django convention for foreign key model fields is to not include the `_id` portion of the name in the field name. This is because when you access that field, you get an actual model instance of the linked model, and not just the ID.

> It’s suggested, but not required, that the name of a [ForeignKey](https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.ForeignKey) field (manufacturer in the example above) be the name of the model, lowercase.

(<https://docs.djangoproject.com/en/5.1/topics/db/models/#many-to-one-relationships>)